cartographer/mapping/map_builder.cc

```diff
@@ -177,8 +177,24 @@ void MapBuilder::LoadMap(io::ProtoStreamReader* const reader) {
   proto::SparsePoseGraph pose_graph;
   CHECK(reader->ReadProto(&pose_graph));
 
-  const int map_trajectory_id = AddTrajectoryForDeserialization();
-  sparse_pose_graph_->FreezeTrajectory(map_trajectory_id);
+  std::map<int, int> trajectory_remapping;
+  for (auto& trajectory_proto : *pose_graph.mutable_trajectory()) {
+    const int new_trajectory_id = AddTrajectoryForDeserialization();
+    // Also takes care of checking for duplicates.
+    CHECK(trajectory_remapping
+              .emplace(trajectory_proto.trajectory_id(), new_trajectory_id)
+              .second);
+    sparse_pose_graph_->FreezeTrajectory(new_trajectory_id);
+    trajectory_proto.set_trajectory_id(new_trajectory_id);
+  }
+
+  // Apply the calculated remapping to constraints in the SparsePoseGraph proto
```

Comment:
Remove this code? The constraints are not used in this function.

Reply:
This is preparation for #544.

Reply:
Then let's do this change there. It is difficult to discuss this change without context and it is orthogonal to the other content of this PR.